File: src/2025/CSES/Intro/repetitions.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    string input;
    cin >> input;

    int maxLength = 0;
    int l = 0;
    for (int r = 0; r < input.size(); r++) {
        if (input[r] != input[l]) {
            maxLength = max(maxLength, r - l);
            l = r;
            continue;
        }

        if (r == input.size() - 1) {
            maxLength = max(maxLength, r - l + 1);
        }
    }
    cout << maxLength << endl;
}

File: src/2025/WarRoom/CsesBattles/cses-rust/src/bin/repetitions.rs
use cses_rust::{read_single_element};
fn main() {
    let word = read_single_element::<String>();
    let mut max_length = 1;
    let mut count = 1;
    let mut prev_char = 'z';

    for ch in word.chars() {
        if ch == prev_char {
            count += 1;
        } else {
            count = 1;
        }

        prev_char = ch;
        max_length = std::cmp::max(max_length, count);
    }

    println!("{}", max_length);
}